### PR TITLE
Add GET /_cluster/master endpoint

### PR DESCRIPTION
**Pull request by DaveCTurner**
We discussed recently that the cluster state API should be considered
"internal" and therefore our usual cast-iron stability guarantees do not hold
for this API.

However, today there is no stable API for identifying the current master node
in JSON format. The REST tests that need to do this will typically call `GET
/_cluster/state` and extract the master from the response.

This change introduces the `GET /_cluster/master` API endpoint to identify the
current master node so that clients do not need to extract the current master
from the cluster state.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/cluster.master/10_basic.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+"get cluster master":
+  - skip:
+      version: " - 7.99.99"
+      reason: Get cluster master API was introduced in 8.0
```
Comment (DaveCTurner) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cluster.master/10_basic.yml]: After backporting, as a follow up, this `skip` can be removed, and all the REST tests that get the master ID from the cluster state can migrate to using this API instead.

**Comment by javanna** [comment]
nice, shall we add support for this new api to the high-level client as part of this PR?

**Comment by ywelsch** [comment]
> However, today there is no stable API for identifying the current master node
in JSON format

I wonder if we can just achieve the same thing with `GET /_nodes/_master?metrics=`

**Comment by DaveCTurner** [comment]
> I wonder if we can just achieve the same thing with `GET /_nodes/_master?metrics=`

The node ID only appears as the (unique) key of the resulting map of nodes. I sketched a possible way to do this in #40052, but it's kinda icky.

**Comment by ywelsch** [comment]
The advantage of the approach in #40052 is that the yml tests become simpler, avoiding the extra call to determine the master at the beginning of the test. I'm not sure yet whether we want to enforce uniqueness or whether we should just support a wildcard selector `*` as a (non-deterministic) choice operator, and leave it to the caller to make sure that the given property makes sense for any key.

**Comment by DaveCTurner** [comment]
In fact most of the REST tests don't want the master node ID: picking an arbitrary ID (perhaps of a data node) is closer to what they want. Closing this in favour of #40052.

**Pull request closed without merging**